### PR TITLE
Update Next.js template

### PR DESCRIPTION
**Pull request by robinmetral**
## Purpose

Updates the `@sumup/cna-template` package.

## Approach and changes

- Finished renaming to `cna-template` (`create-sumup-next-app` was the old package name and now creates confusion when running workspace commands)
- Updated the template, starting from a fresh Next.js default template (Next.js 12, React 18). Added support for the latest versions of Emotion, Jest, and Testing Library
- Updated the template's pages and components for Circuit UI v5

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

**Review by connor-baer — APPROVED** [review]

File: packages/cna-template/.gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@ __reports__
 template/yarn.lock
 template/tsconfig.json
 template/next-env.d.ts
+template-legacy/
```
[the remark this answers is not in the document]
Comment (robinmetral) [reply packages/cna-template/.gitignore:8]: Oopsies, leftover

[the remark this answers is not in the document]
Comment (robinmetral) [reply packages/cna-template/.gitignore:8]: At least I didn't call it "broken-old-bad-template", that would have been unprofessional

[the remark this answers is not in the document]
Comment (robinmetral) [reply packages/cna-template/.gitignore:8]: 😂

**Review by connor-baer — APPROVED** [review]
✨

**Pull request merged**